packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -4,14 +4,32 @@
 
 import 'dart:async';
 
+import 'package:meta/meta.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
 import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
 import '../globals.dart' as globals;
 import '../runner/flutter_command.dart';
 import 'analyze_continuously.dart';
 import 'analyze_once.dart';
 
 class AnalyzeCommand extends FlutterCommand {
-  AnalyzeCommand({bool verboseHelp = false, this.workingDirectory}) {
+  AnalyzeCommand({
+    bool verboseHelp = false,
+    this.workingDirectory,
+    @required FileSystem fileSystem,
```

Comment:
I wonder if we shouldn't bundle these (and maybe some other things from base/) together in a `System` or `Base` object of some sort. That might also simplify testing since the test support code could supply one or more defaults, like `System.test()` or `Base.test()` named constructors with optional named parameters where a different one is needed.